R-package/tests/testthat/test_Predictor.R

```diff
@@ -111,3 +113,117 @@ test_that("start_iteration works correctly", {
     pred_leaf2 <- predict(bst, test$data, start_iteration = 0L, num_iteration = end_iter + 1L, predleaf = TRUE)
     expect_equal(pred_leaf1, pred_leaf2)
 })
+
+test_that("predictions keep row names from the data", {
```

Comment:
these tests look great, thank you! To speed up debugging if they break due to changes in future PRs, can you please break them into 3 test cases?

* "predict() keeps row names from data (binary classification)"
* "predict() keeps row names from data (multi-class classification)"
* "predict() keeps row names from data (regression)"

To avoid duplicating test helper code and to make it clear that those methods were written just for these tests, please move them out of the test case and rename them to `.expect_has_row_names()` and `.expect_doesnt_have_row_names()`.